### ISSUE TITLE
[Bug][A11Y] Radio button not correctly read by screen-readers

### ISSUE THREAD

**Issue by pya35** [Orange-OpenSource/ouds-flutter#362]
### Prerequisites

- [x] I have [searched the backlog](https://github.com/Orange-OpenSource/ouds-flutter/issues) for duplicate or closed feature requests

### Your test device

S22 and iPhone 14+

### OS Version

.

### App version

0.5.0 and 0.6.0

### Describe the issue

On Android: 

- radio button is read "button" if selected and "unlabelled" if not selected
- disabled or error status are not read 

On iOS:
- radio button is read "image" if selected and nothing read if not selected
- disabled or error status are not read 

### Expected Behavior

add correct label and status to radio button

**Comment by pya35** [comment]
linked to #48

**Comment by nouha06** [comment]
@pya35 can you provide me the correct label and status to radio button ?

**Comment by pya35** [comment]
@nouha06 Examples of how to manage a radio button: 
- **Android**:  https://github.com/cvs-health/android-compose-accessibility-techniques/blob/main/doc/components/RadioButtonGroups.md
- **iOS**: https://github.com/cvs-health/ios-swiftui-accessibility-techniques/blob/main/iOSswiftUIa11yTechniques/Documentation/RadioButtons.md

**Comment by B3nz01d** [comment]
@nouha06 is this ready to test? if yes it does not have a build to test

**Comment by nouha06** [comment]
📣 New Firebase App Distribution Alpha upload available 🚀 (Android)

Please, get the new build from App Tester app, or contact the project maintainers (@Tayebsed93) to get access.

Display name: DesignToolbox (alpha) (443)
Version: 0.6.0
Build type: Alpha
Build details (GitHub): 424
cc @B3nz01d (product owner)
cc @Tayebsed93 @nouha06 @AhmedAmineZr (dev team)
cc @jerome-regnier (design team)
cc @pya35 (accessibility team)

**Comment by nouha06** [comment]
📣 New TestFlight Alpha upload available 🚀 (iOS)

Please, get the new build from TestFlight app, or contact the project maintainers (@Tayebsed93) to get access.

Display name: Design Toolbox Flutter (443)
Version: 0.6.0
Build type: Alpha
Build details (GitHub): 424
cc @B3nz01d (product owner)
cc @Tayebsed93 @AhmedAmineZr @nouha06 (dev team)
cc @jerome-regnier (design team)
cc @pya35 (accessibility team)

**Comment by pya35** [comment]
@nouha06 Regarding selected/not selected value, are you using the label or the role to give the information? For both iOS and Android, when the radio is disabled or read-only we hear twice the selected state with 2 different ways, one from the real state of component, the other one from the label I suppose ?:

<img width="281" height="44" alt="Image" src="https://github.com/user-attachments/assets/dc8d431f-5530-4e49-8822-454af88cb711" />

**Comment by nouha06** [comment]
> [@nouha06](https://github.com/nouha06) Regarding selected/not selected value, are you using the label or the role to give the information? For both iOS and Android, when the radio is disabled or read-only we hear twice the selected state with 2 different ways, one from the real state of component, the other one from the label I suppose ?:
> 
> <img alt="Image" width="281" height="44" src="https://private-user-images.githubusercontent.com/105045667/505943584-dc8d431f-5530-4e49-8822-454af88cb711.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjE1NzE4NjAsIm5iZiI6MTc2MTU3MTU2MCwicGF0aCI6Ii8xMDUwNDU2NjcvNTA1OTQzNTg0LWRjOGQ0MzFmLTU1MzAtNGU0OS04ODIyLTQ1NGFmODhjYjcxMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMDI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTAyN1QxMzI2MDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jZTcyNzdjYjlhNGEwNTJkNWY1NjIyNDc5NjBiMWExNWRiMDQxNGUxNDU2MGE4NzNjYzU4NzA1YmUyMmVlMmJkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Bl-wl26pwyY-eqlS_AomTmkmWKVgwc2MGvDNwbf4MCU">
Yes, I’m using a custom label, and I’ve already fixed this issue. However, we currently have a deployment problem for iOS on TestFlight, so I’ll deliver this fix later.

**Comment by nouha06** [comment]
Added blocking labels as deployment to TestFlight is not possible at the moment.

**Comment by Tayebsed93** [comment]
The issue has been fixed and the ticket has been closed, as we cannot release a new version. That’s why the TestFlight blocking label was applied.

**Comment by B3nz01d** [comment]
Closed during Sprint # 23

**Comment by nouha06** [comment]
📣 New Firebase App Distribution BETA upload available 🚀 (Android)

Please, get the new build from App Tester app, or contact the project maintainers (@Tayebsed93) to get access.

Display name: DesignToolbox (BETA) (485)
Version: 0.6.0
Build type: BETA
Build details (GitHub): 424
cc @B3nz01d (product owner)
cc @Tayebsed93 @AhmedAmineZr @nouha06 (dev team)
cc @jerome-regnier (design team)
cc @pya35 (accessibility team)

**Comment by nouha06** [comment]
New TestFlight BETA upload available 🚀 (iOS)

Please, get the new build from TestFlight app, or contact the project maintainers (@Tayebsed93) to get access.

Display name: Design Toolbox Flutter (485)
Version: 0.6.0
Build type: BETA
Build details (GitHub): 424
cc @B3nz01d (product owner)
cc @Tayebsed93 @AhmedAmineZr @nouha06 (dev team)
cc @jerome-regnier (design team)
cc @pya35 (accessibility team)
